java/Lexer.java

```diff
@@ -17,55 +20,48 @@ public Lexer(String input) {
     public Token nextToken() {
         this.skipWhitespace();
 
-        Token t = null;
-        TokenType tt = TokenType.ILLEGAL;
-        switch (this.getCc()) {
-            case '{' -> tt = TokenType.LSQIRLY;
-            case '}' -> tt = TokenType.RSQIRLY;
-            case '(' -> tt = TokenType.LPAREN;
-            case ')' -> tt = TokenType.RPAREN;
-            case ',' -> tt = TokenType.COMMA;
-            case ';' -> tt = TokenType.SEMI;
-            case '+' -> tt = TokenType.PLUS;
-            case '=' -> tt = TokenType.EQUAL;
-            case '\0' -> tt = TokenType.EOF;
-        }
+        final var tokenType = switch (this.getCc()) {
+            case '{' -> TokenType.LSQIRLY;
+            case '}' -> TokenType.RSQIRLY;
+            case '(' -> TokenType.LPAREN;
+            case ')' -> TokenType.RPAREN;
+            case ',' -> TokenType.COMMA;
+            case ';' -> TokenType.SEMI;
+            case '+' -> TokenType.PLUS;
+            case '=' -> TokenType.EQUAL;
+            case '\0' -> TokenType.EOF;
+            default -> TokenType.ILLEGAL;
+        };
 
-        if (tt.equals(TokenType.EOF)) {
-            t = this.createToken(tt, "eof");
-        } else {
-            t = this.createToken(tt, String.valueOf(this.getCc()));
-        }
+        final var token = switch (tokenType) {
+            case EOF -> this.createToken(tokenType, "eof");
+            default -> this.createToken(tokenType, String.valueOf(this.getCc()));
+        };
+
+        if (Character.isLetter(this.getCc())) {
+            var ident = this.indent();
```

Comment:
I dislike the use of auto or var, as i understand var works like auto and determines the variables type by itself right? [source](https://docs.oracle.com/en/java/javase/20/language/local-variable-type-inference.html) Not really necessary here because its just a string.

Reply:
The main reason I used var here is to showcase it. Not many people are familiar with modern Java and probably still think about the 1.7 days. It could show its a possibility.

Reply:
I'm okay with ``var`` but ``final var`` is unnecessary for me. I personally thinks, you don't need to declare a variable as final in method scope. Since your method is supposed to be small, hence it should be obvious how the variable is being used.

Reply:
@andirady I removed final from local variables. It does cause some unnecessary noise.